inc/dbmysqliterator.class.php

```diff
@@ -215,7 +215,16 @@ function buildQuery ($table, $crit = "", $log = false) {
                               break;
                         }
                         foreach ($val as $jointable => $joincrit) {
-                           $join .= " $jointype JOIN " .  DBmysql::quoteName($jointable) . " ON (" . $this->analyseCrit($joincrit) . ")";
+                           $joinExpressionKeys = ['FKEY' => null, 'ON' => null];
```

Comment:
Might be an opportuniy to have  this array in a private property and mutualize it with this liine 
https://github.com/glpi-project/glpi/blob/9.3/bugfixes/inc/dbmysqliterator.class.php#L490

Reply:
This should be factorized, yes.

I'm do not know yet if I prefer your prévious proposal or this one :D But I'm globally OK with the change, thank you.

Reply:
the new one avoids the ugly key "SUBQUERY". This is smoother to write, and seems to me more natural and easier to learn.

Reply:
Building queries from an array is not natural :D